src/table/utils/handle-key-press.js

```diff
@@ -133,46 +132,49 @@ export const bodyHandleKeyPress = ({
   evt,
   rootElement,
   cellCoord,
-  selectionState,
   cell,
   selectionDispatch,
   isAnalysisMode,
   setFocusedCellCoord,
   announce,
   keyboard,
+  selectionsAPI = null,
 }) => {
+  const isInSelectionMode = selectionsAPI?.isModal();
+
   switch (evt.key) {
     case 'ArrowUp':
     case 'ArrowDown':
     case 'ArrowRight':
     case 'ArrowLeft': {
-      !isCtrlShift(evt) && moveFocus(evt, rootElement, cellCoord, selectionState, setFocusedCellCoord, announce);
+      !isCtrlShift(evt) && moveFocus(evt, rootElement, cellCoord, setFocusedCellCoord, announce, isInSelectionMode);
       break;
     }
     // Space bar: Selects value.
     case ' ': {
       preventDefaultBehavior(evt);
-      cell?.isDim && isAnalysisMode && selectCell({ selectionState, cell, selectionDispatch, evt, announce });
+      cell?.isDim && isAnalysisMode && selectionDispatch({ type: 'select', payload: { cell, evt, announce } });
```

Comment:
why sometimes `selectCell` and sometimes `selectionDispatch({ type: 'select', payload: { cell, evt, announce } })`?

Reply:
`selectCell` is only called inside the reducer. That logic could live in the reducer itself, but I kept the separate function to not clutter the reducer